relay-server/src/actors/processor.rs

```diff
@@ -1152,7 +1152,13 @@ impl EnvelopeProcessorService {
                 if !profiling_enabled {
                     return ItemAction::DropSilently;
                 }
-                match relay_profiling::expand_profile(&item.payload(), state.event.value()) {
+                // If we don't have an event at this stage, we need to drop the profile.
+                let event = if let Some(event) = state.event.value() {
+                    event
+                } else {
+                    return ItemAction::DropSilently;
+                };
```

Comment:
```suggestion
                let Some(event) = state.event.value() else {
                    return ItemAction::DropSilently;
                };
```
I think you can do a `let..else` here